src/horizontal-pod-autoscaler.ts

```diff
@@ -0,0 +1,865 @@
+      type: MetricSourceType.PODS,
+      pods: {
+        metric: {
+          name: options.name,
+          selector: options?.labelSelector?._toKube(),
+        },
+        target: options.target._toKube(),
+      },
+    });
+  }
+
+  /**
+   * Tracks the available memory of a target.
+   *
+   * Use case:
+   * * Scale up when CPU is above 40%.
+   */
+  public static resourceCpu(target: MetricTarget): Metric {
+    return new Metric({
+      type: MetricSourceType.RESOURCE,
+      resource: {
+        name: ResourceType.CPU,
+        target: target._toKube(),
+      },
+    });
+  }
+
+  /**
+   * Tracks the available CPU of a target.
+   *
+   * Use case:
+   * * Scale up when Memory is above 512MB.
+   */
+  public static resourceMemory(target: MetricTarget): Metric {
+    return new Metric({
+      type: MetricSourceType.RESOURCE,
+      resource: {
+        name: ResourceType.MEMORY,
+        target: target._toKube(),
+      },
+    });
+  }
+
+  /**
+   * Tracks the available Storage of a target.
+   */
+  public static resourceStorage(target: MetricTarget): Metric {
+    return new Metric({
+      type: MetricSourceType.RESOURCE,
+      resource: {
+        name: ResourceType.STORAGE,
+        target: target._toKube(),
+      },
+    });
+  }
+
+  /**
+   * Tracks the available Ephemeral Storage of a target.
+   */
+  public static resourceEphemeralStorage(target: MetricTarget): Metric {
+    return new Metric({
+      type: MetricSourceType.RESOURCE,
+      resource: {
+        name: ResourceType.EPHEMERAL_STORAGE,
+        target: target._toKube(),
+      },
+    });
+  }
+
+  public readonly type: string;
+  private constructor(private readonly metric: k8s.MetricSpecV2) {
+    this.type = metric.type;
+  }
+
+  /**
+   * @internal
+   */
+  public _toKube(): k8s.MetricSpecV2 {
+    return this.metric;
+  }
+
+}
+
+export enum ResourceType {
+  /**
+   * CPU
+   */
+  CPU= 'cpu',
+  /**
+   * Memory
+   */
+  MEMORY = 'memory',
+  /**
+   * Volume size
+   */
+  STORAGE = 'storage',
+  /**
+   * Local ephemeral storage
+   */
+  EPHEMERAL_STORAGE = 'ephemeral-storage',
+}
+
+
+/**
+ * How to interpret the metric value.
+ */
+export enum MetricTargetType {
+  /**
+   * The target value must be exactly the same as the metric value.
+   */
+  VALUE = 'Value',
+  /**
+   * The target value is an average metric value measured across all relevant pods.
+   */
+  AVERAGE_VALUE = 'AverageValue',
+  /**
+   * The target value is a percentage of the metric measured across all relevant pods.
+   */
+  UTILIZATION = 'Utilization',
+}
+
+
+/**
+ * A metric condition that will trigger scaling behavior when satisfied.
+ *
+ * @example
+ *
+ * MetricTarget.averageUtilization(70); // 70% average utilization
+ *
+ */
+export class MetricTarget {
+  /**
+  * Target a specific target value.
+  *
+  * @param value The target value. Floats can be passed within a string. i.g. `"1.2"`
+  */
+  public static value(value: string): MetricTarget {
```

Comment:
You can pass floats to a `number` type...

```suggestion
  public static value(value: number): MetricTarget {
```